GitCommands/RevisionReader.cs

```diff
@@ -451,17 +439,54 @@ DateTime ParseUnixDateTime()
             return true;
         }
 
+        [CanBeNull]
+        private static (string body, string additionalData) ParseCommitBody([NotNull] IStringLineReader reader, [NotNull] string subject)
+        {
+            int lengthOfSubjectRepeatedInBody = subject.Length + 2/*newlines*/;
+            if (reader.Remaining == lengthOfSubjectRepeatedInBody + EndOfBody.Length)
+            {
+                return (body: subject, additionalData: null);
+            }
+
+            string tail = reader.ReadToEnd() ?? "";
+            int indexOfEndOfBody = tail.LastIndexOf(EndOfBody, StringComparison.InvariantCulture);
+            if (indexOfEndOfBody < 0)
+            {
+                // TODO log this parse error
+                Debug.Fail("Missing end-of-body marker in the log -- this should not happen");
+                return (body: null, additionalData: null);
+            }
+
+            string additionalData = null;
+            if (tail.Length > indexOfEndOfBody + EndOfBody.Length)
+            {
+                additionalData = tail.Substring(indexOfEndOfBody + EndOfBody.Length).TrimStart();
+            }
+
+            string body = indexOfEndOfBody == lengthOfSubjectRepeatedInBody ? subject : tail.Substring(0, indexOfEndOfBody).TrimEnd();
+            return (body, additionalData);
```

Comment:
Will this give another allocation of the string? If so GC will run more often during the startup, something that @drewnoakes looked into. I do not see any difference in the Linux repo when testing with a stopwatch.

Reply:
No, the intention of this code is to avoid another allocation of a string for single-line commit messages - as before the regression.